homeassistant/components/shelly/utils.py

```diff
@@ -298,7 +298,7 @@ def get_rpc_channel_name(device: RpcDevice, key: str) -> str:
         entity_name = device.config[key].get("name", device_name)
 
     if entity_name is None:
-        if [k for k in key if k.startswith(("input", "switch"))]:
+        if key.startswith(("input", "switch")):
```

Comment:
Should we use split_entity_id here so this only matches input platforms we want?